### PR TITLE
Don't build on Appveyor/Travis when changing Travis/Appveyor configuration

### PR DESCRIPTION
**Pull request by sahitpj**
Fixes issue #207

### CODE REVIEW

**Comment by marco-c** [comment]
Thanks for the patch! The same should be done for the other way around (when appveyor.yml is modified we shouldn't run Travis).
It looks like there's an error in the current patch though: `Error parsing appveyor.yml: Invalid "skip_commits" setting: file (Line: 6, Column: 3)`.

**Comment by sahitpj** [comment]
Hey @marco-c will look into that immediately.

**Comment by sahitpj** [comment]
screenshot 2018-10-25 16 13 57

hey @marco-c , it seems that only one build is failing, which I think that has something to do with another issue

**Comment by calixteman** [comment]
@sahitpj you're right it's unrelated. So don't care about that.

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,10 @@
 sudo: false
 language: rust
+
+skip_commits:
```
Comment (marco-c) [inline .travis.yml]: Are you sure that this is supported on Travis? I wasn't able to find any reference to this. If it isn't supported, then you'll have to do something like https://github.com/facebook/react/pull/2000/files.

**Comment by sahitpj** [comment]
True, but I was wondering why `travis` didn't throw an error. Yeah I will implement that then @marco-c

**Comment by sahitpj** [comment]
Hey @marco-c I think this should do

File: .travis.yml [hunk 2]
```diff
@@ -1,5 +1,16 @@
 sudo: false
 language: rust
+
+before_install:
```
Comment (marco-c) [inline .travis.yml]: Could you move this after `matrix` and before `install`? It's more coherent, as before_install happens on Travis right before install.

File: .travis.yml [hunk 3]
```diff
@@ -1,5 +1,16 @@
...
+- |
+    if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then
+      TRAVIS_COMMIT_RANGE="FETCH_HEAD...$TRAVIS_BRANCH"
+    fi
+    git diff --name-only $TRAVIS_COMMIT_RANGE | grep -qvE '(.appveyor.yml)' || {
```
Comment (marco-c) [inline .travis.yml]: The appveyor file is `appveyor.yml`, not `.appveyor.yml`

File: .travis.yml (hunk 2 above)
Comment (marco-c) [reply .travis.yml]: Looks good, but this change is still missing @sahitpj

File: .travis.yml [hunk 4]
```diff
@@ -1,5 +1,6 @@
 sudo: false
 language: rust
+
```
Comment (marco-c) [inline .travis.yml]: Nit: no newline here

File: .travis.yml [hunk 5]
```diff
@@ -103,7 +103,15 @@ matrix:
         - CLANG_CXX=clang++
         - LLVM_CONFIG=llvm-config
         - DEPLOY=1
-
```
Comment (marco-c) [inline .travis.yml]: Nit: don't remove newline here :P

As a general rule, try to avoid unrelated whitespace changes (don't add/remove newlines in parts of the code you're not modifying), as they make the vcs history needlessly more complex.

This is OK though, don't worry about it (when a review comment starts with "Nit:" it means that it is a minor comment that can be also be ignored).

Comment (sahitpj) [reply .travis.yml]: Will remember that, thanks a lot 😄
